### PR TITLE
Clarify failure scenarios 

### LINKED ISSUE #306: Clarify failure scenarios
We should clarify the different failure scenarios that are currently formulated in a slightly "overlapping" way [here](https://docs.fiskaltrust.cloud/docs/poscreators/middleware-doc/general/cash-register-integration/failure-scenarios).

In fact, there are IMO 3 scenarios to cover:
- **SCU not reachable**: Middleware goes into failed mode/circuit breaker mode, returns the ftState `0x02` -> Can be resolved via a zero receipt
- **Queue/MW not reachable**: No response from the Middleware, e.g. due to network or MW outage -> Receipts should be re-sent as soon as the MW is reachable again, using the flag `0x0000000000010000` to switch to late-signing mode. When the receipts are re-sent, failure-mode needs to be ended with a zero receipt
- **Cash register outage**: The whole cash register is not working anymore (e.g. because of a power outage), and the local regulations require to track handwritten receipts during that time -> Receipts should be re-sent to the Middleware after the systems are back online with the flag `0x0000000000080000`

### PR DESCRIPTION
**Pull request by TSchmiedlechner**
Fixes #306

### CODE REVIEW

File: doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md [hunk 1]
```diff
@@ -6,7 +6,31 @@ title: Failure scenarios
...
+2. **Middleware not reachable or failing**: When the cash register doesn't receive any response from the Middleware (e.g. because of a network or server outage), receipts _must_ be re-sent as soon as the Middleware is reachable again, using the flag `0x0000000000010000` to switch to the _late-signing mode_. If required in the respective country, the Middleware will automatically take care of late-signig the receipts then. When the receipts have been successfully re-sent, failure-mode _must_ be left by sending a _zero receipt_.
+3. **Cash register outage**: When the whole cash register/POS system is not working anymore (e.g. because of a power outage or a system failure) and the local fiscalization laws require the tracking of handwritten receipts (like e.g. in Germany) in that case, receipts _may_ be re-sent to the Middleware after the systems are back online with the _handwritten ftReceiptCaseFlag_ `0x0000000000080000`. This is optional, but recommended for consistentency in case of an audit.
```
Comment (StefanKert) [inline doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md:12]: I think that we cou
IMO we should leave the optional part out.

```suggestion
3. **Cash register outage**: When the whole cash register/POS system is not working anymore (e.g. because of a power outage or a system failure) and the local fiscalization laws require the tracking of handwritten receipts (like e.g. in Germany) in that case, receipts _may_ be re-sent to the Middleware after the systems are back online with the _handwritten ftReceiptCaseFlag_ `0x0000000000080000`. 
```

File: doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md [hunk 2]
```diff
@@ -6,7 +6,31 @@ title: Failure scenarios
...
+
+#### Signature Cration Unit not reachable or failing
+If the communication to the SCU fails (e.g. when the secure Signature Creation Device is not reachable), the POS System can continue to operate until the SCU is accessible again. Receipts created in a state where no communication is possible with the SCU are protected by the security mechanism of fiskaltrust. The fiskaltrust.Middleware will respond with the ftState = `0x02` "SCU communication failed". The POS-System receives the response and processes the data it contains. For following Requests no more communication attempts are done to avoid long waiting times for each Receipt request/Receipt response sequence.
```
Comment (StefanKert) [inline doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md:15]: Should we be more precise on the state here? SInce the state ends with 0x02 and is more or less a very long number

Comment (StefanKert) [inline doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md:15]: Maybe we should also focus on the fact that the Middleware does all things necessary to still be compliant instead of writing that the communication is not performed anymore. I guess if we have a market that we would need to check communication we would probably still do so. 

TL;DR let's focus on the middleware capabilities that this mode keeps the receipt chain compliant by still allowing to handle outages of the local signing device

File: doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md [hunk 3]
```diff
@@ -6,7 +6,31 @@ title: Failure scenarios
...
+</p>
+
+![no-scu-connection](./images/10-no-scu-connection.png)
+  
+When the SCU is reachable again, a Zero-Receipt must be sent, which forces a communication retry towards the SSCD. If the fiskaltrust.Middleware is able to connect to the SCU again, the ftState = 0x00 (ok) is returned to the POS system via the response and the fiskaltrust.Middleware is ready for normal operation again. Furthermore, the response contains a listing of the requests that were not signed by the SSCD. The requests affected by the failure of the communication with the SCU do not have to be sent to the Queue again after the problem has been resolved.
+
+:::tip
+
+We recommend to make the ZeroReceipt after a failure a manual operation, and not automatically send it via the POS system as soon as a failure state is returned. In most scenarios, only Operators can determine if the connection to the SSCD can be re-established, e.g. when the internet or the device is reconnected. Automatically sending zero-receipts might lead to unnecessary wait times if the connection can't be established at this point in time.
```
Comment (StefanKert) [inline doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md:26]: Should we also add a "tip" to print the signatures? Because in this mode the possystem is required to still print the signatures. We could add a section to reference the suggestion to directly print sigantures because they would still be good in such a case.

File: doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md [hunk 4]
```diff
@@ -6,7 +6,31 @@ title: Failure scenarios
...
+
+:::
+
+![reestablished-scu-connection](./images/11-reestablished-connection.png)
+
+
+#### Middleware not reachable or failing
```
Comment (StefanKert) [inline doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md:33]: Maybe we should also mention that in case of a connection issue after sending a receipt possystems should send a request flag to double check if it was processed already to make sure we don't process revenue twice.

File: doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md (hunk 2 above)
Comment (TSchmiedlechner) [reply doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md]: I had the feeling that this was already "mentioned" in a way, but you're right, makes more sense to be explicit about this - done

File: doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md (hunk 3 above)
Comment (TSchmiedlechner) [reply doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md:26]: Done

File: doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md (hunk 4 above)
Comment (TSchmiedlechner) [reply doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md:39]: Done, thanks

File: doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md (hunk 3 above)
Comment (TSchmiedlechner) [reply doc/general/cash-register-integration/cash-register-integration-failure-scenarios.md:26]: Done 👍🏻

**Review by Bluette-C-Riviere — APPROVED** [review]
